airflow/utils/helpers.py

```diff
@@ -305,6 +306,9 @@ def on_terminate(p):
     except OSError as err:
         if err.errno == errno.ESRCH:
             return
+        # If operation not permitted error is thrown due to run_as_user, use sudo to kill the process
+        if err.errno == errno.EPERM:
+            subprocess.check_call(["sudo", "-n", "-" + str(sig), str(os.getpgid(pid))])
```

Comment:
```suggestion
            subprocess.check_call(["sudo", "--non-interactive", "-" + str(sig), str(os.getpgid(pid))])
```

Please use long-form arguments in code, it's much easier  to understand what it does then.

Other than that this will be okay

Reply:
But, sudo --non-interactive doesn't work on all Linux machines. I will try again and will show the error message

Reply:
But, sudo --non-interactive doesn't work on all Linux machines. I will try again and will show the error message

Reply:
Does it not? It should be part of the posix spec I thought.

Reply:
Oh yes, I was wrong. Can you then add a comment saying "-n = non-interactive" please?

Reply:
yes, added comment to indicate that it's non-interactive